### PR TITLE
Split ReactDataGrid and Addons into separate packages

### PR DESCRIPTION
**Pull request by willgates**
Continuing from #351

### CODE REVIEW

File: config/webpack.react-data-grid.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  },  
+  externals: {
+    "react": {
+      root : 'React',
+      commonjs : 'react',
+      commonjs2 : 'react',
+      amd : 'react'
+    },
+    "react-dom": {
+			root : 'ReactDOM',
```
Comment (diogofcunha) [inline config/webpack.react-data-grid.js]: Format this one

File: examples/scripts/documentation01-gettingstarted.js [hunk 2]
```diff
@@ -26,9 +26,9 @@ module.exports = React.createClass({
         <div className="code-block js">
           <pre>&lt;script src="//fb.me/react-0.14.6.js"&gt;&lt;/script&gt;<br/>&lt;script type="text/javascript" src="react-data-grid/dist/react-data-grid.js"&gt;&lt;/script&gt;</pre>
         </div>
-        <p>Or use react-data-grid-with-addons.js to use advanced features</p>
+        <p>Or use react-data-grid.ui-plugins.js to use advanced features</p>
```
Comment (diogofcunha) [inline examples/scripts/documentation01-gettingstarted.js]: For the sake of consistency shouldn't it be:

> And include react-data-grid.ui-plugins.js to use advanced features

?

File: gulp/tasks/examples.js [hunk 3]
```diff
@@ -63,17 +62,17 @@ gulp.task("copy-dist", ['dist'], function(){
 });
 
 gulp.task('minify-examples', ['copy-dist'], function(){
-	  gulp.src('./examples/build/react-data-grid-with-addons.js')
-    .pipe(uglify())
-    .pipe(rename('react-data-grid-with-addons.min.js'))
-    .pipe(gulp.dest('./examples/build'))
-    .on('error', gutil.log);
+	//   gulp.src('./examples/build/react-data-grid-with-addons.js')
```
Comment (diogofcunha) [inline gulp/tasks/examples.js]: Delete commented code.

File: gulp/tasks/examples.js [hunk 4]
```diff
@@ -63,17 +62,17 @@ gulp.task("copy-dist", ['dist'], function(){
...
+    // .pipe(uglify())
+    // .pipe(rename('react-data-grid-with-addons.min.js'))
+    // .pipe(gulp.dest('./examples/build'))
+    // .on('error', gutil.log);
 
-    gulp.src('./examples/build/react-data-grid.js')
-    .pipe(uglify())
-    .pipe(rename('react-data-grid.min.js'))
-    .pipe(gulp.dest('./examples/build'))
-    .on('error', gutil.log);
```
Comment (diogofcunha) [inline gulp/tasks/examples.js]: Same as in the previous comment

File: gulp/tasks/examples.js [hunk 5]
```diff
@@ -23,9 +23,8 @@ webpackConfig = {
 	resolve: {
 		alias: {
 			react: path.resolve('./node_modules/react'),
-      'react-dom': path.resolve('./node_modules/react-dom'),
```
Comment (diogofcunha) [inline gulp/tasks/examples.js]: I know it wasn't you but can you format this spacing?

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "react-data-grid",
-  "version": "0.14.46",
+  "version": "0.15.0",
```
Comment (diogofcunha) [inline package.json]: I see that we will do a major version bump, it is absolutely necessary, was concerned about it, good job :+1:

File: src/RowsContainer.js [hunk 7]
```diff
@@ -1,14 +1,44 @@
...
+    super();
+    this.state = {ContextMenuContainer: this.getContextMenuContainer()};
+    this.hasContextMenu = this.hasContextMenu.bind(this);
+    this.renderRowsWithContextMenu = this.renderRowsWithContextMenu.bind(this);
+    this.getContextMenuContainer = this.getContextMenuContainer.bind(this);
+  }
+
+  getContextMenuContainer() {
+    if (!(window.ReactDataGridPlugins && window.ReactDataGridPlugins)) {
+      throw new ('You need to include ReactDataGrid UiPlugins in order to initialise context menu');
```
Comment (diogofcunha) [inline src/RowsContainer.js]: Lets throw a new Error [source](http://stackoverflow.com/questions/9156176/what-is-the-difference-between-throw-new-error-and-throw-someobject-in-javas) instead

File: src/RowsContainer.js [hunk 8]
```diff
@@ -1,14 +1,44 @@
...
+  constructor() {
+    super();
+    this.state = {ContextMenuContainer: this.getContextMenuContainer()};
+    this.hasContextMenu = this.hasContextMenu.bind(this);
+    this.renderRowsWithContextMenu = this.renderRowsWithContextMenu.bind(this);
+    this.getContextMenuContainer = this.getContextMenuContainer.bind(this);
+  }
+
+  getContextMenuContainer() {
+    if (!(window.ReactDataGridPlugins && window.ReactDataGridPlugins)) {
```
Comment (diogofcunha) [inline src/RowsContainer.js]: Why are we checking the same thing two times?

File: package.json (hunk 6 above)
Comment (diogofcunha) [inline package.json]: After discussion We agreed that version should bump to 1.0.0 and not 0.15.0, as we will not be following the same version as react anymore it wouldn't make much sense to keep this kind of versioning.

**Comment by diogofcunha** [comment]
We are missing vital information about the update, people must be notified about the changes in the way we export things from now on.

Let's update the README containing relevant information for 1.0.0, I will also write some release notes in 1.0.0 tag after release.

**Comment by diogofcunha** [comment]
Apart from the comments made in the code LGTM. 
Tested on my local env.

File: README.md [hunk 9]
```diff
@@ -20,9 +20,27 @@ installed from npm.
 
 Overview 
 --------
-
 ReactDataGrid is an advanced JavaScript spreadsheet component built using React
 
+Exported Modules
```
Comment (diogofcunha) [inline README.md]: @bakesteve can you review the English on this description?

Comment (bakesteve) [inline README.md]: I've updated it. can you verify the instructions are clear / correct, esp on migration

**Pull request merged**